ethereum/spec/src/main/java/tech/pegasys/teku/spec/ForkManifest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import tech.pegasys.teku.datastructures.state.Fork;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.constants.SpecConstants;
+import tech.pegasys.teku.ssz.SSZTypes.Bytes4;
+
+public class ForkManifest {
+  private final List<Fork> forkSchedule;
+
+  public ForkManifest(SpecConstants genesisConstants) {
```

Comment:
Assuming we make this constructor private, accepting a list of forks (suggestion below), we could add some more validation here to make sure the forks are consistent:
* Ensure that each `fork.previousVersion == prevFork.currentVersion`
* Ensure all `fork.epoch` values are unique
* Ensure we have a genesis fork
* Ensure `fork.currentVersion != fork.previousVersion` except for the genesis fork

Also, I don't suppose the list of forks is large, but we could keep them in sorted order in a `NavigableMap` (epoch -> fork).  Then we can query for an epoch's fork with `NavigableMap.floorEntry`.